src/app/ReadClient.cpp

```diff
@@ -418,22 +416,36 @@ CHIP_ERROR ReadClient::OnMessageReceived(Messaging::ExchangeContext * apExchange
     }
     else if (aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::StatusResponse))
     {
-        VerifyOrExit(apExchangeContext == mpExchangeCtx, err = CHIP_ERROR_INCORRECT_STATE);
-        err = StatusResponse::ProcessStatusResponse(std::move(aPayload));
-        SuccessOrExit(err);
+        CHIP_ERROR statusError = CHIP_NO_ERROR;
```

Comment:
This call flow here results in us eventually sending a StatusResponse below in the call to `ResponseMessageHandled` right? Why would we send a status response in response to a status response?

Reply:
Previously in this PR, I am sending a status response in response to status response, After chatting with boris, we think if malformed status response comes, we still need to send status response since the issue responsibility is in other side.